### PR TITLE
Send --help output to stdout

### LINKED ISSUE #4878: "cargo help test" output should go to stdout
Currently, `cargo help test` output goes to stderr.  This makes it unnecessarily difficult to pipe the output through a pager.  Furthermore, this isn't an error situation -- the command exists to print the help, so it seems to me that it ought to go to stdout.  (If it's any use, this is a longstanding GNU convention for tools as well; so many if not most core Linux utilities work this way.)

### PR DESCRIPTION
**Pull request by tromey**
Fixes #4878

### CODE REVIEW

**Comment by alexcrichton** [comment]
Thanks! Looks like some tests may be failing though?

**Comment by tromey** [comment]
> error[E0277]: the trait bound `for<'de> Flags: serde::Deserialize<'de>` is not satisfied

I don't understand how this could be caused by my patch.

**Comment by matklad** [comment]
@bors try

Looks like the failure was caused by a yanked docopt from the first of January 😆

File: src/cargo/lib.rs [hunk 1]
```diff
@@ -142,7 +142,8 @@ pub fn exit_with_error(err: CliError, shell: &mut Shell) -> ! {
         } else if fatal {
             drop(shell.error(&error))
         } else {
-            drop(writeln!(shell.err(), "{}", error))
+            // Non-fatal messages, like --help output, go to stdout.
```
Comment (matklad) [inline src/cargo/lib.rs]: I believe this comment is a bit misleading: a ton of non-fatal messages, like `Downloading foo...` purposefully use stderr. Maybe drop this comment or change it to "unlike all other human-readable output, help goes to stdout so as to make it easier to use pager with cargo --help".

**Comment by alexcrichton** [comment]
Oops sorry about that! Good catch!

File: src/cargo/lib.rs (hunk 1 above)
Comment (tromey) [reply src/cargo/lib.rs]: Makes sense.  I'll just drop it, I suppose anybody who is curious can read the history.

**Pull request merged**